power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/serializer.hpp

```diff
@@ -495,17 +518,61 @@ class Serializer {
         packer_.pack_map(static_cast<uint32_t>(count));
     }
 
-    static bool check_nan(RawElementPtr element_ptr, MetaAttribute const& attribute) {
-        return ctype_func_selector(attribute.ctype, [element_ptr, &attribute]<class T> {
-            return is_nan(attribute.get_attribute<T const>(element_ptr));
+    static bool check_nan(BufferView const& element_buffer, MetaComponent const& component,
+                          MetaAttribute const& attribute) {
+        if (element_buffer.buffer->data != nullptr) {
+            RawElementPtr element_ptr = component.advance_ptr(element_buffer.buffer->data, element_buffer.idx);
+            return ctype_func_selector(attribute.ctype, [element_ptr, &attribute]<class T> {
+                return is_nan(attribute.get_attribute<T const>(element_ptr));
+            });
+        } else if (auto it = std::ranges::find_if(element_buffer.buffer->attributes,
+                                                  [&attribute](auto const& attribute_buffer) {
+                                                      return attribute_buffer.meta_attribute == &attribute;
+                                                  });
```

Comment:
this is a potentially expensive search in the hot loop. Unfortunately, I have tried many different things but haven't been able to come up with a satisfactory solution except for making 2 completely separate control flows (one for row-based, one for columnar).

@TonyXiang8787 what do you think?